packages/create-llama/e2e/basic.spec.ts

```diff
@@ -42,16 +46,40 @@ for (const templateType of templateTypes) {
             templateEngine,
             templateUI,
             appType,
+            externalPort,
           );
 
-          const port = Math.floor(Math.random() * 10000) + 10000;
-          const cps = await runApp(cwd, name, appType, port);
+          const cps = await runApp(cwd, name, appType, port, externalPort);
 
           // test frontend
           if (appType !== "--no-frontend") {
             await page.goto(`http://localhost:${port}`);
             await expect(page.getByText("Built by LlamaIndex")).toBeVisible();
           }
+
+          // test backend
+          const backendPort =
```

Comment:
@thucpn after adding the FE test, we can keep this test for testing the backend for no-frontend cases